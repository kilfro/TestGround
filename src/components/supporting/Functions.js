/**
 * Создает новый уникальный для данного массива 'id'
 *
 * @param array         массив, элементы коготоро имеют поле 'id'
 * @returns {number}    новый ID
 */
export const getNextId = (array) => {
    if (array.length === 0)
        return 1;

    const maxId = Math.max.apply(Math, array.map(item => item.id));
    return maxId + 1;
};

export const getRadioGroupValue = (options) => {
    let checkedId = -1;

    options.map(op => {
        if (op.isRight) {
            checkedId = op.id;
        }
    });

    return checkedId;
};

export const checkQuestion = (toCheck) => {
    const {question, options} = toCheck;

    if (!question || question === '') {
        return false;
    }

    const emptyOptions = options.filter(op => op.text === '');
    if (emptyOptions.length !== 0) {
        return false;
    }

    const correctAnswers = options.filter(op => op.isRight === true);
    if (correctAnswers.length === 0) {
        return false;
    }

    return true;
};