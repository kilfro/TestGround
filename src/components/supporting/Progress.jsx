import {lighten, withStyles} from '@material-ui/core/styles';
import {LinearProgress} from "@material-ui/core";

export const Progress = withStyles({
    root: {
        height: 10,
        backgroundColor: lighten('rgba(0, 0, 0, 0.38)', 0.5),
        borderRadius: 20
    },
    bar: {
        borderRadius: 20,
        backgroundColor: '#3f51b5'
    },
})(LinearProgress);