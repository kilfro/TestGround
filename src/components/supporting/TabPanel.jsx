import React from 'react';

const TabPanel = (props) => {
    const {children, tabPosition, index, ...other} = props;

    return (
        <div
            role="tabpanel"
            hidden={tabPosition !== index}
            id={`full-width-tabpanel-${index}`}
            aria-labelledby={`full-width-tab-${index}`}
            {...other}
        >
            {tabPosition === index && (<div>{children}</div>)}
        </div>
    );
};

export default TabPanel;