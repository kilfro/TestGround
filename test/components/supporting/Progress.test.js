import React from 'react';
import {Progress} from "../../../src/components/supporting/Progress";

describe('Progress', () => {
    it('should render correctly', () => {
        const component = mount(<Progress variant='determinate' value={50}/>);
        expect(component).toMatchSnapshot();
    });
});