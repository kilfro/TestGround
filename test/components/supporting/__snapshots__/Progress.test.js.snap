// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render correctly 1`] = `
<WithStyles(WithStyles(ForwardRef(LinearProgress)))
  value={50}
  variant="determinate"
>
  <WithStyles(ForwardRef(LinearProgress))
    classes={
      Object {
        "bar": "WithStyles(ForwardRef(LinearProgress))-bar-2",
        "root": "WithStyles(ForwardRef(LinearProgress))-root-1",
      }
    }
    value={50}
    variant="determinate"
  >
    <ForwardRef(LinearProgress)
      classes={
        Object {
          "bar": "MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-2",
          "bar1Buffer": "MuiLinearProgress-bar1Buffer",
          "bar1Determinate": "MuiLinearProgress-bar1Determinate",
          "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate",
          "bar2Buffer": "MuiLinearProgress-bar2Buffer",
          "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate",
          "barColorPrimary": "MuiLinearProgress-barColorPrimary",
          "barColorSecondary": "MuiLinearProgress-barColorSecondary",
          "buffer": "MuiLinearProgress-buffer",
          "colorPrimary": "MuiLinearProgress-colorPrimary",
          "colorSecondary": "MuiLinearProgress-colorSecondary",
          "dashed": "MuiLinearProgress-dashed",
          "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary",
          "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary",
          "determinate": "MuiLinearProgress-determinate",
          "indeterminate": "MuiLinearProgress-indeterminate",
          "query": "MuiLinearProgress-query",
          "root": "MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-1",
        }
      }
      value={50}
      variant="determinate"
    >
      <div
        aria-valuenow={50}
        className="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-1 MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-2 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-50%)",
            }
          }
        />
      </div>
    </ForwardRef(LinearProgress)>
  </WithStyles(ForwardRef(LinearProgress))>
</WithStyles(WithStyles(ForwardRef(LinearProgress)))>
`;
